s3-source-connector/src/test/java/io/aiven/kafka/connect/s3/source/input/AvroTransformerTest.java

```diff
@@ -73,46 +74,67 @@ void testConfigureValueConverter() {
     void testReadAvroRecordsInvalidData() {
         final InputStream inputStream = new ByteArrayInputStream("mock-avro-data".getBytes(StandardCharsets.UTF_8));
 
-        final DatumReader<GenericRecord> datumReader = new GenericDatumReader<>();
-        final List<Object> records = avroTransformer.readAvroRecords(inputStream, datumReader);
+        final Stream<Object> records = avroTransformer.getRecords(() -> inputStream, "", 0, s3SourceConfig, 0);
 
-        assertThat(records.size()).isEqualTo(0);
+        final List<Object> recs = records.collect(Collectors.toList());
+        assertThat(recs).isEmpty();
     }
 
     @Test
     void testReadAvroRecords() throws Exception {
-        final ByteArrayOutputStream avroData = generateMockAvroData();
+        final ByteArrayOutputStream avroData = generateMockAvroData(25);
         final InputStream inputStream = new ByteArrayInputStream(avroData.toByteArray());
 
-        final DatumReader<GenericRecord> datumReader = new GenericDatumReader<>();
-        final List<Object> records = avroTransformer.readAvroRecords(inputStream, datumReader);
+        final Stream<Object> records = avroTransformer.getRecords(() -> inputStream, "", 0, s3SourceConfig, 0);
 
-        assertThat(records.size()).isEqualTo(2);
+        final List<Object> recs = records.collect(Collectors.toList());
+        assertThat(recs).hasSize(25);
     }
 
-    ByteArrayOutputStream generateMockAvroData() throws IOException {
+    @Test
+    void testReadAvroRecordsSkipFew() throws Exception {
```

Comment:
Can you add a test where more records are skipped then exist in the stream?
See what that behaviour looks like, as if a user does some manual intervention after hitting a malformed record or something the numbers might not add up properly.

Reply:
Good one.